### PR TITLE
fix field derive

### LINKED ISSUE #745: Derived fields duplicate enums instead type cast
Derived field
```xml
<field derivedFrom="RELOAD">
  <name>IDR</name>
  <description>Selects, if Reload Register number is incremented, decremented or not modified</description>
  <bitRange>[27:26]</bitRange>
  <access>read-write</access>
</field>
```

```rust
#[doc = "Field `IDR` reader - Selects, if Reload Register number is incremented, decremented or not modified"]
pub type IDR_R = crate::FieldReader<IDR_A>;
#[doc = "Selects, if Reload Register number is incremented, decremented or not modified\n\nValue on reset: 0"]
#[derive(Clone, Copy, Debug, PartialEq, Eq)]
#[repr(u8)]
pub enum IDR_A {
    #[doc = "0: Selects Reload Register number 0"]
    RELOAD0 = 0,
    #[doc = "1: Selects Reload Register number 1"]
    RELOAD1 = 1,
    #[doc = "2: Selects Reload Register number 2"]
    RELOAD2 = 2,
    #[doc = "3: Selects Reload Register number 3"]
    RELOAD3 = 3,
}
```

Derived enum

```xml
<field>
  <name>IDR</name>
  <description>Selects, if Reload Register number is incremented, decremented or not modified</description>
  <bitRange>[27:26]</bitRange>
  <access>read-write</access>
  <enumeratedValues derivedFrom="RELOAD" />
</field>
```

```rust
#[doc = "Field `IDR` reader - Selects, if Reload Register number is incremented, decremented or not modified"]
pub use RELOAD_R as IDR_R;
#[doc = "Field `IDR` writer - Selects, if Reload Register number is incremented, decremented or not modified"]
pub use RELOAD_W as IDR_W;
```

https://github.com/rust-embedded/svdtools/pull/159

### PR DESCRIPTION
**Pull request by burrbull**
partially fixes #745 

need more tests

cc @kossnikita

### CODE REVIEW

**Comment by kossnikita** [comment]
Doesn't work with unnamed enumeratedValues

https://github.com/rust-embedded/svd2rust/blob/e3087cceac23e5075b2ddfa5e4cef65a7e873e55/src/generate/register.rs#L604C32-L604C32

File: src/generate/register.rs [hunk 1]
```diff
@@ -601,6 +600,8 @@ pub fn fields(
             let dpath = ev.derived_from.take();
             if let Some(dpath) = dpath {
                 epath = Some(derive_enumerated_values(&mut ev, &dpath, &fpath, index)?);
+            } else if let Some(path) = fdpath.as_ref() {
+                epath = Some(path.new_enum(ev.name.clone().unwrap()));
```
Comment (kossnikita) [inline src/generate/register.rs:604]: Doesn't work with unnamed enumeratedValues

Comment (burrbull) [reply src/generate/register.rs]: try again please

File: src/generate/register.rs [hunk 2]
```diff
@@ -601,6 +600,8 @@ pub fn fields(
             let dpath = ev.derived_from.take();
             if let Some(dpath) = dpath {
                 epath = Some(derive_enumerated_values(&mut ev, &dpath, &fpath, index)?);
+            } else if let Some(path) = fdpath.as_ref() {
+                epath = Some(path.new_enum(ev.name.clone().unwrap_or_else(|| f.name.clone())));
             }
             // TODO: remove this hack
             if let Some(epath) = epath.as_ref() {
```
Comment (kossnikita) [inline src/generate/register.rs:613]: Still panics here

```
panicked at 'called `Option::unwrap()` on a `None` value', src\generate\register.rs:608:45
```

Comment (burrbull) [reply src/generate/register.rs:613]: and again

Comment (kossnikita) [reply src/generate/register.rs:613]: Not in your code, below it

Comment (kossnikita) [reply src/generate/register.rs:613]: tested with
```
svd2rust 0.30.0 (8a44606 2023-10-01)
```

```
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', src\generate\register.rs:614:45
```

**Comment by kossnikita** [comment]
In my case it works good

File: src/generate/register.rs (hunk 2 above)
Comment (burrbull) [reply src/generate/register.rs]: what now?

**Comment by burrbull** [comment]
> In my case it works good

Could you test yet once, please.

**Comment by kossnikita** [comment]
Looks good to me

```xml
<field derivedFrom="RELOAD">
  <name>IDR</name>
  <description>Selects, if Reload Register number is incremented, decremented or not modified</description>
  <bitRange>[27:26]</bitRange>
  <access>read-write</access>
</field>
```

```rust
#[doc = "Field `IDR` reader - Selects, if Reload Register number is incremented, decremented or not modified"]
pub use RELOAD_R as IDR_R;
#[doc = "Field `IDR` writer - Selects, if Reload Register number is incremented, decremented or not modified"]
pub use RELOAD_W as IDR_W;
```

```
svd2rust 0.30.0 (ee3afce 2023-10-01)
```

**Review by Emilgardis — APPROVED** [review]

**Pull request merged**